### PR TITLE
#215 add new custom hooks

### PR DESCRIPTION
**Pull request by dylan-pham**
I've updated the code implementation but still have to update the tests.

### CODE REVIEW

File: src/containers/projects-table/projects-table.tsx [hunk 1]
```diff
@@ -3,54 +3,48 @@
...
-      })
-    );
+  const { isLoading, errorMessage, responseData } = useAllProjects();
+
+  /**
+   * Transform given list of projects into a list of display projects.
+   * @param projects list of projects to transform
+   * @returns list of display projects
+   */
+  const updateData = (projects: Project[]) => {
```
Comment (jamescd18) [inline src/containers/projects-table/projects-table.tsx:20]: I might suggest a better name for this function like `projectsToDisplayProjects()` or `transformToDisplayProjects()`

File: src/containers/projects-table/projects-table.tsx [hunk 2]
```diff
@@ -3,54 +3,48 @@
...
+      <>
+        <h3>Oops, sorry!</h3>
+        <h5>There was an error loading the page.</h5>
+        <p>{errorMessage ? errorMessage : 'The data did not load properly.'}</p>
+      </>
+    );
+  }
 
-  return <PrjsTable allProjects={allProjects} />;
+  return <PrjsTable allProjects={updateData(responseData)!} />;
```
Comment (jamescd18) [inline src/containers/projects-table/projects-table.tsx:47]: Any idea why the ! is on the right side of the closing parentheses? I'm not quite familiar enough / maybe not remembering. Does it not work putting the ! to the left of the closing parentheses?

**Review by jamescd18 — CHANGES_REQUESTED** [review]
Looking good! These behavior changes look to be correct. I left a couple suggestions / curiosities.

File: src/containers/projects-table/projects-table.tsx (hunk 2 above)
Comment (dylan-pham) [reply src/containers/projects-table/projects-table.tsx:47]: I believe including the ! on the right side is telling the compiler that the return value of the function call is non-null whereas including it on the left side just means that responseData is non-null. I honestly hadn't seen this syntax before so I had to do some googling.

**Pull request closed without merging**